torch/csrc/api/src/data/init.cpp

```diff
@@ -0,0 +1,479 @@
+      .def_readwrite(
+          "batch_size", &ChunkDatasetOptions::batch_size_, "Batch size")
+      .def_readwrite(
+          "cache_size",
+          &ChunkDatasetOptions::cache_size_,
+          "Cache size to be preloaded before batching");
+
+  /// Sampler
+  py::class_<Sampler<>, std::shared_ptr<Sampler<>>, PySampler<>> sampler(
+      data, "Sampler");
+  sampler.def(py::init<>(), "Create and return a new `Sampler` instance");
+  sampler.def(
+      "reset",
+      &Sampler<>::reset,
+      "Resets the `Sampler`'s internal state. Typically called before a new epoch. Optionally, accepts a new size when reseting the sampler",
+      py::arg("new_size"));
+  sampler.def(
+      "next",
+      &Sampler<>::next,
+      "Returns the next index if possible, or an empty optional if the sampler is exhausted for this epoch",
+      py::arg("batch_size"));
+  sampler.def(
+      "save",
+      &Sampler<>::save,
+      "Serializes the `RandomSampler` to the `archive`",
+      py::arg("archive"));
+  sampler.def(
+      "load",
+      &Sampler<>::load,
+      "Deserializes the `Sampler` from the `archive`",
+      py::arg("archive"));
+
+  /// SamplerWrapper
+  py::class_<SamplerWrapper, std::shared_ptr<SamplerWrapper>> sampler_wrapper(
+      data, "SamplerWrapper", sampler);
+  sampler_wrapper.def(
+      py::init<std::shared_ptr<Sampler<>>, size_t>(),
+      "Create and return a new `SamplerWrapper` instance",
+      py::arg("sampler"),
+      py::arg("stride") = 0);
+  sampler_wrapper.def(
+      "reset",
+      &SamplerWrapper::reset,
+      "Resets the `Sampler`'s internal state. Typically called before a new epoch. Optionally, accepts a new size when reseting the sampler",
+      py::arg("new_size"));
+  sampler_wrapper.def(
+      "next",
+      &SamplerWrapper::next,
+      "Returns the next index if possible, or an empty optional if the sampler is exhausted for this epoch",
+      py::arg("batch_size"));
+  sampler_wrapper.def(
+      "save",
+      &SamplerWrapper::save,
+      "Serializes the `RandomSampler` to the `archive`",
+      py::arg("archive"));
+  sampler_wrapper.def(
+      "load",
+      &SamplerWrapper::load,
+      "Deserializes the `Sampler` from the `archive`",
+      py::arg("archive"));
+  sampler_wrapper.def(
+      "set_current_stride",
+      &SamplerWrapper::set_current_stride,
+      "Sets the underlying sampler to a new stride",
+      py::arg("stride"));
+  sampler_wrapper.def(
+      "set_total_strides",
+      &SamplerWrapper::set_total_strides,
+      "Sets the total number of strides for the underlying sampler",
+      py::arg("total"));
+  sampler_wrapper.def(
+      "stride",
+      &SamplerWrapper::current_stride,
+      "Returns current sampler stride");
+  sampler_wrapper.def(
+      "total_strides",
+      &SamplerWrapper::total_strides,
+      "Returns the total number of sampler strides");
+
+  /// DistributedSampler
+  py::class_<DistributedSampler<>, PyDistributedSampler<>> distributed_sampler(
```

Comment:
There is a `DistributedSampler` in Python, so why are we binding it in here??

Reply:
ChunkDataset is a C++ API (exposed to python) and it only accepts C++ samplers (exposed to python). We can't use a Python sampler on a C++ Dataset